File: schemas/marketplace/drizzle/vendor_profiles.ts
// vendor_profiles: Public-facing storefront information and policies for each vendor.
// See README.md for full design rationale.

import {
  pgTable,
  uuid,
  text,
  timestamp,
  jsonb,
} from "drizzle-orm/pg-core";
import { vendors } from "./vendors";

export const vendorProfiles = pgTable(
  "vendor_profiles",
  {
    id: uuid("id").primaryKey().defaultRandom(),
    vendorId: uuid("vendor_id")
      .notNull()
      .unique()
      .references(() => vendors.id, { onDelete: "cascade" }),
    displayName: text("display_name").notNull(),
    tagline: text("tagline"),
    description: text("description"),
    logoUrl: text("logo_url"),
    bannerUrl: text("banner_url"),
    websiteUrl: text("website_url"),
    socialLinks: jsonb("social_links"),
    returnPolicy: text("return_policy"),
    shippingPolicy: text("shipping_policy"),
    createdAt: timestamp("created_at", { withTimezone: true })
      .notNull()
      .defaultNow(),
    updatedAt: timestamp("updated_at", { withTimezone: true })
      .notNull()
      .defaultNow()
      .$onUpdateFn(() => new Date()),
  }
);

File: schemas/marketplace/spacetimedb/vendor_profiles.rs
// vendor_profiles: Extended profile information for vendor storefronts.
// See README.md for full design rationale.

use spacetimedb::Timestamp;

#[spacetimedb::table(name = vendor_profiles, public)]
pub struct VendorProfile {
    #[primary_key]
    pub id: String, // UUID

    #[unique]
    pub vendor_id: String, // UUID, FK -> vendors.id (cascade delete)

    pub display_name: String,
    pub tagline: Option<String>,
    pub description: Option<String>,
    pub logo_url: Option<String>,
    pub banner_url: Option<String>,
    pub website_url: Option<String>,
    pub social_links: Option<String>, // JSON
    pub return_policy: Option<String>,
    pub shipping_policy: Option<String>,
    pub created_at: Timestamp,
    pub updated_at: Timestamp,
}
